Can't click on 'Workloads' group header to go to Workloads

This is a bug specific to group headers in the side nav.

Go to the local cluster and click on Workloads.

Click on CronJobs.

Now try and click on Workloads - not by clicking on the text, but in the area around it (see red line below) - although the cursor shows a hand icon and although this works on the first click into Workloads, this does not work when the group is expanded - you have to click on the text.

![Screenshot 2024-03-05 at 12 00 15](https://github.com/rancher/dashboard/assets/1955897/5eec0975-cc32-4693-a928-32c554c06095)

You should be able to click anywhere in the sider nav group header to go back to it.

This e2e test is failing in Jenkins and also locally with this error: 

AssertionError: Timed out retrying after 60000ms: 
expected 'https://jenkins-k3iz1ru1.qa.rancher.space/dashboard/c/local/explorer/workload' 
to equal 
'https://jenkins-k3iz1ru1.qa.rancher.space/dashboard/c/local/explorer#cluster-events'
      + expected - actual

      -'https://jenkins-k3iz1ru1.qa.rancher.space/dashboard/c/local/explorer/workload'
      +'https://jenkins-k3iz1ru1.qa.rancher.space/dashboard/c/local/explorer#cluster-events'

It seems there needs to be some delay after clicking 'Workloads'. This should do the trick.


_____
import WorkloadPagePo from '@/cypress/e2e/po/pages/explorer/workloads.po.ts';
    
const workloads = new WorkloadPagePo();
workloads.waitForPage();
____

--- a/cypress/e2e/tests/navigation/side-nav/product-side-nav.spec.ts
+++ b/cypress/e2e/tests/navigation/side-nav/product-side-nav.spec.ts
@@ -1,6 +1,7 @@
 import HomePagePo from '@/cypress/e2e/po/pages/home.po';
 import BurgerMenuPo from '@/cypress/e2e/po/side-bars/burger-side-menu.po';
 import ProductNavPo from '@/cypress/e2e/po/side-bars/product-side-nav.po';
+import WorkloadPagePo from '@/cypress/e2e/po/pages/explorer/workloads.po';
 
 Cypress.config();
 describe('Side navigation: Cluster ', { tags: ['@navigation', '@adminUser'] }, () => {
@@ -84,15 +85,16 @@ describe('Side navigation: Cluster ', { tags: ['@navigation', '@adminUser'] }, (
     const group = productNavPo.groups().eq(1); // First group is 'Workloads'
 
     // Select and expand current top-level group
-    group.click().then((workloadsGroup) => {
-      const workloadsUrl = (workloadsGroup.find('a')[0] as HTMLAnchorElement).href;
+    group.click();
+    const workloads = new WorkloadPagePo();
 
+    workloads.waitForPage();
+
+    cy.url().then((workloadsUrl) => {
       // Go to the second subgroup
       productNavPo.visibleNavTypes().eq(1).click({ force: true });
       // Clicking back should take us back to workloads
-      productNavPo.tabHeaders().eq(1).click(1, 1).then(() => {
-        cy.url().should('equal', workloadsUrl);
-      });
+      productNavPo.tabHeaders().eq(1).click(1, 1).then(() => cy.url().should('equal', workloadsUrl));
     });
   });
 });

--- a/cypress/e2e/tests/navigation/side-nav/product-side-nav.spec.ts
+++ b/cypress/e2e/tests/navigation/side-nav/product-side-nav.spec.ts
@@ -84,13 +84,15 @@ describe('Side navigation: Cluster ', { tags: ['@navigation', '@adminUser'] }, (
     const group = productNavPo.groups().eq(1); // First group is 'Workloads'
 
     // Select and expand current top-level group
-    group.click();
+    group.click().then((workloadsGroup) => {
+      const workloadsUrl = (workloadsGroup.find('a')[0] as HTMLAnchorElement).href;
 
-    cy.url().then((workloadsUrl) => {
       // Go to the second subgroup
       productNavPo.visibleNavTypes().eq(1).click({ force: true });
       // Clicking back should take us back to workloads
-      productNavPo.tabHeaders().eq(1).click(1, 1).then(() => cy.url().should('equal', workloadsUrl));
+      productNavPo.tabHeaders().eq(1).click(1, 1).then(() => {
+        cy.url().should('equal', workloadsUrl);
+      });
     });
   });
 });

